### PR TITLE
Support add and modify column via table rewrites (new format only for now)

### PR DESCRIPTION
**Pull request by zachmu**

### CODE REVIEW

**Comment by zachmu** [comment]
This relies on https://github.com/dolthub/go-mysql-server/pull/1004 to actually do anything useful

File: go/libraries/doltcore/doltdb/root_val.go [hunk 1]
```diff
@@ -470,7 +470,9 @@ func (root *RootValue) GenerateTagsForNewColumns(
 	if headRoot != nil {
 		for _, col := range existingCols {
 			for i := range newColNames {
-				if strings.ToLower(newColNames[i]) == strings.ToLower(col.Name) {
+				// Only re-use tags if the noms kind didn't change
+				if strings.ToLower(newColNames[i]) == strings.ToLower(col.Name) &&
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/doltdb/root_val.go:474]: you should add a todo here to revisit this

File: go/libraries/doltcore/sqle/globalstate/auto_increment_tracker.go [hunk 2]
```diff
@@ -46,15 +46,30 @@ func CoerceAutoIncrementValue(val interface{}) (uint64, error) {
 	return val.(uint64), nil
 }
 
-func NewAutoIncrementTracker(ctx context.Context, ws *doltdb.WorkingSet) (ait AutoIncrementTracker, err error) {
-	ait = AutoIncrementTracker{
+// AutoIncrementTracker knows how to track the latest auto-increment values for every table across all sessions.
+type AutoIncrementTracker interface {
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/globalstate/auto_increment_tracker.go:50]: we don't need this interface, it only has one implementation

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM!

**Pull request merged**